File: SCMS.Mobile/lib/src/features/patients/domain/patient_models.dart
class PatientProfileResponse {
  const PatientProfileResponse({
    required this.patientId,
    required this.userId,
    required this.name,
    this.mobileNo,
    this.email,
    this.dateOfBirth,
    this.gender,
    this.bloodType,
    this.actualAddress,
    this.allergies,
    this.chronicConditions,
    this.pastSurgeries,
    this.familyHistory,
    this.vaccinationHistory,
    required this.createdAt,
  });

  factory PatientProfileResponse.fromJson(Map<String, dynamic> json) {
    return PatientProfileResponse(
      patientId: json['patientId'] as int? ?? 0,
      userId: json['userId'] as int? ?? 0,
      name: json['name'] as String? ?? '',
      mobileNo: json['mobileNo'] as String?,
      email: json['email'] as String?,
      dateOfBirth: json['dateOfBirth'] as String?,
      gender: json['gender'] as String?,
      bloodType: json['bloodType'] as String?,
      actualAddress: json['actualAddress'] as String?,
      allergies: json['allergies'] as String?,
      chronicConditions: json['chronicConditions'] as String?,
      pastSurgeries: json['pastSurgeries'] as String?,
      familyHistory: json['familyHistory'] as String?,
      vaccinationHistory: json['vaccinationHistory'] as String?,
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'] as String)
          : DateTime.now(),
    );
  }

  final int patientId;
  final int userId;
  final String name;
  final String? mobileNo;
  final String? email;
  final String? dateOfBirth;
  final String? gender;
  final String? bloodType;
  final String? actualAddress;
  final String? allergies;
  final String? chronicConditions;
  final String? pastSurgeries;
  final String? familyHistory;
  final String? vaccinationHistory;
  final DateTime createdAt;
}

class PatientProfileRequest {
  const PatientProfileRequest({
    required this.name,
    this.mobileNo,
    this.email,
    this.dateOfBirth,
    this.gender,
    this.bloodType,
    this.actualAddress,
    this.allergies,
    this.chronicConditions,
    this.pastSurgeries,
    this.familyHistory,
    this.vaccinationHistory,
  });

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'mobileNo': mobileNo,
      'email': email,
      'dateOfBirth': dateOfBirth,
      'gender': gender,
      'bloodType': bloodType,
      'actualAddress': actualAddress,
      'allergies': allergies,
      'chronicConditions': chronicConditions,
      'pastSurgeries': pastSurgeries,
      'familyHistory': familyHistory,
      'vaccinationHistory': vaccinationHistory,
    };
  }

  final String name;
  final String? mobileNo;
  final String? email;
  final String? dateOfBirth;
  final String? gender;
  final String? bloodType;
  final String? actualAddress;
  final String? allergies;
  final String? chronicConditions;
  final String? pastSurgeries;
  final String? familyHistory;
  final String? vaccinationHistory;
}

File: SCMS.Domain/Features/Patients/Models/PatientModels.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using SCMS.Shared;

namespace SCMS.Domain.Features.Patients.Models
{
    /// <summary>Request parameters for listing patient profiles with pagination.</summary>
    public class GetPatientProfilesRequest : PaginationRequest
    {
    }

    /// <summary>Response item for listing patient profiles.</summary>
    public sealed record GetPatientProfilesResponse
    {
        public int PatientId { get; init; }
        public int UserId { get; init; }
        public string Name { get; init; } = null!;
        public string? MobileNo { get; init; }
        public string? Email { get; init; }
        public DateOnly? DateOfBirth { get; init; }
        public string? Gender { get; init; }
        public string? BloodType { get; init; }
        public string? ActualAddress { get; init; }
        public string? Allergies { get; init; }
        public string? ChronicConditions { get; init; }
        public string? PastSurgeries { get; init; }
        public string? FamilyHistory { get; init; }
        public string? VaccinationHistory { get; init; }
        public DateTime CreatedAt { get; init; }
    }

    /// <summary>Request parameters for searching patient profiles by keyword.</summary>
    public class SearchPatientProfilesRequest : PaginationRequest
    {
        [Required(ErrorMessage = "Search query is required.")]
        public string Query { get; set; } = string.Empty;
    }

    /// <summary>Response item for searching patient profiles.</summary>
    public sealed record SearchPatientProfilesResponse
    {
        public int PatientId { get; init; }
        public int UserId { get; init; }
        public string Name { get; init; } = null!;
        public string? MobileNo { get; init; }
        public string? Email { get; init; }
        public DateOnly? DateOfBirth { get; init; }
        public string? Gender { get; init; }
        public string? BloodType { get; init; }
        public string? ActualAddress { get; init; }
        public string? Allergies { get; init; }
        public string? ChronicConditions { get; init; }
        public string? PastSurgeries { get; init; }
        public string? FamilyHistory { get; init; }
        public string? VaccinationHistory { get; init; }
        public DateTime CreatedAt { get; init; }
    }

    /// <summary>Payload for creating a patient profile.</summary>
    public sealed record CreatePatientProfileRequest
    {
        [Required(ErrorMessage = "Patient name is required.")]
        public required string Name { get; init; }

        public string? MobileNo { get; init; }
        public string? Email { get; init; }
        public DateOnly? DateOfBirth { get; init; }
        public string? Gender { get; init; }
        public string? BloodType { get; init; }
        public string? ActualAddress { get; init; }

        public string? Allergies { get; init; }
        public string? ChronicConditions { get; init; }
        public string? PastSurgeries { get; init; }
        public string? FamilyHistory { get; init; }
        public string? VaccinationHistory { get; init; }
    }

    /// <summary>Response returned upon creating a patient profile.</summary>
    public sealed record CreatePatientProfileResponse
    {
        public int PatientId { get; init; }
        public int UserId { get; init; }
        public string Name { get; init; } = null!;
        public string? MobileNo { get; init; }
        public string? Email { get; init; }
        public DateOnly? DateOfBirth { get; init; }
        public string? Gender { get; init; }
        public string? BloodType { get; init; }
        public string? ActualAddress { get; init; }
        public string? Allergies { get; init; }
        public string? ChronicConditions { get; init; }
        public string? PastSurgeries { get; init; }
        public string? FamilyHistory { get; init; }
        public string? VaccinationHistory { get; init; }
        public DateTime CreatedAt { get; init; }
    }

    /// <summary>Detailed patient profile response.</summary>
    public sealed record GetPatientProfileByIdResponse
    {
        public int PatientId { get; init; }
        public int UserId { get; init; }
        public string Name { get; init; } = null!;
        public string? MobileNo { get; init; }
        public string? Email { get; init; }
        public DateOnly? DateOfBirth { get; init; }
        public string? Gender { get; init; }
        public string? BloodType { get; init; }
        public string? ActualAddress { get; init; }
        public string? Allergies { get; init; }
        public string? ChronicConditions { get; init; }
        public string? PastSurgeries { get; init; }
        public string? FamilyHistory { get; init; }
        public string? VaccinationHistory { get; init; }
        public DateTime CreatedAt { get; init; }
    }

    // Backward-compatibility models
    public class PatientProfilesRequest : PaginationRequest
    {
        public string? Search { get; set; }
    }

    public sealed record PatientProfileRequest
    {
        [Required(ErrorMessage = "Patient name is required.")]
        public required string Name { get; init; }

        public string? MobileNo { get; init; }
        public string? Email { get; init; }
        public DateOnly? DateOfBirth { get; init; }
        public string? Gender { get; init; }
        public string? BloodType { get; init; }
        public string? ActualAddress { get; init; }

        public string? Allergies { get; init; }
        public string? ChronicConditions { get; init; }
        public string? PastSurgeries { get; init; }
        public string? FamilyHistory { get; init; }
        public string? VaccinationHistory { get; init; }
    }

    public sealed record PatientProfileResponse
    {
        public int PatientId { get; init; }
        public int UserId { get; init; }
        public string Name { get; init; } = null!;
        public string? MobileNo { get; init; }
        public string? Email { get; init; }
        public DateOnly? DateOfBirth { get; init; }
        public string? Gender { get; init; }
        public string? BloodType { get; init; }
        public string? ActualAddress { get; init; }

        public string? Allergies { get; init; }
        public string? ChronicConditions { get; init; }
        public string? PastSurgeries { get; init; }
        public string? FamilyHistory { get; init; }
        public string? VaccinationHistory { get; init; }

        public DateTime CreatedAt { get; init; }
    }

    /// <summary>Patient medical and visit history timeline response.</summary>
    public sealed record PatientHistoryResponse
    {
        public int PatientId { get; init; }
        public string PatientName { get; init; } = null!;
        public List<TimelineItemDto> Timeline { get; set; } = new();
    }

    public sealed record TimelineItemDto
    {
        public DateTime Date { get; init; }
        public string Type { get; init; } = null!;
        public string Title { get; init; } = null!;
        public string Description { get; init; } = null!;
        public int LinkedId { get; init; }
    }

    /// <summary>Comprehensive medical summary response for clinical review and exports.</summary>
    public sealed record MedicalSummaryResponse
    {
        public int PatientId { get; init; }
        public string PatientName { get; init; } = null!;
        public DateOnly? DateOfBirth { get; init; }
        public string? Gender { get; init; }
        public string? BloodType { get; init; }
        
        public string? Allergies { get; init; }
        public string? ChronicConditions { get; init; }
        public string? PastSurgeries { get; init; }
        public string? FamilyHistory { get; init; }
        public string? VaccinationHistory { get; init; }

        public List<PatientVitalsHistoryDto> VitalsHistory { get; set; } = new();
        public List<ActivePrescriptionSummaryDto> ActivePrescriptions { get; set; } = new();
    }

    public sealed record PatientVitalsHistoryDto
    {
        public DateTime Date { get; init; }
        public double? WeightKg { get; init; }
        public int? BloodPressureSystolic { get; init; }
        public int? BloodPressureDiastolic { get; init; }
        public double? TemperatureC { get; init; }
        public int? PulseBpm { get; init; }
        public int? Spo2Percent { get; init; }
        public double? HeightCm { get; init; }
        public double? Bmi { get; init; }
    }

    public sealed record ActivePrescriptionSummaryDto
    {
        public int PrescriptionId { get; init; }
        public DateTime Date { get; init; }
        public string DiseaseName { get; init; } = null!;
        public List<string> Medicines { get; init; } = new();
    }
}
